nservicebus/architecture/publishing-from-web-applications.md

```diff
@@ -0,0 +1,89 @@
+2. **Web Application Scale-Out** - Because events can only be published from one logical publisher, it can be problematic to scale out a web application that is publishing events. More on this later.
+3. **Web App Recycling** - Prior to IIS 7.5 introducing the AlwaysRunning feature (AKA AlwaysOn), web applications not receiving could be recycled or idled at any time, which would result in incoming messages building up in an input queue, without an application actively processing them.
+
+Because of this, conventional wisdom has suggested that when in the context of a web application, it is better to only send commands to a back-end service endpoint, which can then publish a similar event.
+
+But, things change:
+
+* With IIS 7.5 and later, IIS supports AlwaysRunning mode.
+* IIS is no longer the only way to host web applications, which can now be hosted by [OWIN](http://owin.org/)/[Katana](http://www.asp.net/aspnet/overview/owin-and-katana), [NancyFx](http://nancyfx.org/), [FubuMVC](http://mvc.fubu-project.org/), and others. This largely removes the concern over web application recycling.
+* NServiceBus 4.0 introduced the ability to have transports besides MSMQ. Not all of the transports handle Publish/Subscribe in the same way as MSMQ - some even support it natively.
+* Cloud transports generally have an associated cost per interaction. This means that requiring an unnecessary send before publish costs extra money to operate.
+* In NServiceBus 4.0 and 5.0, the process of creating a send-only endpoint (which is still capable of publishing messages, but does not have an input queue or process incoming messages) was made much easier.
+
+## Publish/Subscribe Mechanics
+
+First it's important to understand how publish/subscribe actually works, consisting of the subscription phase, and then the actual publish operation.
+
+### Subscription
+
+An endpoint must register its interest in a message by subscribing to a specific message type. How this occurs is dependent upon the message transport being used.
+
+Some transports (Azure Service Bus, RabbitMQ) support publish/subscribe natively. This means that when an endpoint wants to subcribe to an event, it contacts the broker directly, and the broker maintains the list of subscriptions.
+
+Other transports (MSMQ, SQL, Azure Storage Queues) do not support native publish/subscribe, and instead use storage-driven publishing with message-driven subscriptions. This means that each endpoint is responsible for maintaining its own subscription storage, usually in a database. When an endpoint wants to subscribe to an event, it sends a subscription request message to the owner endpoint, which will update its own subscription storage.

```

Comment:
Could it read nicer if order changed to be "...and instead use message-driven subscriptions with storage-driven publishing."
It's probably my head insisting on pub/sub order :)